### DISCUSSION TITLE
how to add a footnote using the kableExtra package without a page cut off?

### DISCUSSION THREAD

**Discussion by kmink3225** [quarto-dev/quarto-cli#6337] [Q&A] [answered]
### Description

# Description

I have encountered some outliers in the output of my analysis. As a result, I need to add footnotes below a table to provide a simple description of the outliers. When I use the knitr kable alone, the output tables are well-organized without any page breaks. However, once I load the kableExtra package, numerous page breaks occur. Below is the code provided for reproducibility. If anyone knows how to add footnotes without causing page breaks, I would greatly appreciate your input. It is also acceptable if the kableExtra package is not used.

Thank you in advance!

# reproducible code
````
---
format: pdf
engine: knitr
execute:
  echo: false
  warning: false
---

```{r, knitr_chunk}
library(tidyverse)
library(knitr)
head(iris,50)%>%kable(booktabs=TRUE)
```

```{r, kableExtra_chunk}
library(kableExtra)
head(iris,50)%>%kable(booktabs=TRUE)%>%
footnote(general = "the footnote test" )
```
````

# output

## kable output

![image](https://github.com/quarto-dev/quarto-cli/assets/61118160/cbededb7-22ee-4c4d-b84b-0bc6bee88d87)

## kableExtra output

![image](https://github.com/quarto-dev/quarto-cli/assets/61118160/bea97db7-0172-41fe-aa6c-5f20ef13e07c)

**Comment by mcanouil** [comment]
It seems to be an issue with what `kableExtra` exports for PDF/LaTeX.
I suggest to take a look at the R package documentation and to take a look at the LaTeX code produced by adding `keep-tex: true` to your YAML front matter.

**Comment by cderv** [comment, marked as answer]
One important thing to understand is that `knitr::kable()` will output Markdown by default, unless you specify some options. You can see that by using `keep-md: true` to see the intermediary file. 
When you are using `kableExtra` package, you are forcing HTML or LaTeX table for `kable()`. This is in the [doc](https://cran.r-project.org/web/packages/kableExtra/vignettes/awesome_table_in_pdf.pdf): 
> Starting fromkableExtra0.9.0,whenyouloadthispackage(library(kableExtra)),itwillautomaticallysetup theglobaloption‘knitr.table.format’basedonyourcurrentenvironment.

Doc is here and you should really read it before using **kableExtra** : https://haozhu233.github.io/kableExtra/

Now you understand the difference, you need to understand that Quarto will do some processing to make Tables looks good, and this works for Markdown tables. Doc is here: https://quarto.org/docs/authoring/tables.html
Part of this looks good is using the right LaTeX to split the table on pagebreak. 

When you output a LaTeX table from your R chunk, it will already be formatted by **kableExtra** and we don't parse this Tables. It is left asis. 
So you need to do the formatting yourself. And in your example you did not do any specific formatting on your LaTeX as if you intend to have a split tables, you need to use `longtable`. We talk about that in https://bookdown.org/yihui/rmarkdown-cookbook/kable.html
> Sometimes your table may be longer than a page. In this case, you can use the argument longtable = TRUE, which uses the LaTeX package [longtable](https://ctan.org/pkg/longtable) to span your table to multiple pages.

So just use this as this is how it supposed to work. This would get you the same type of output 
````markdown

```{r, kableExtra_chunk}
library(kableExtra)
head(iris,50) |> kable(booktabs=TRUE, longtable = TRUE) |> footnote(general = "the footnote test" ) |> kable_styling(latex_options = c("repeat_header"))
```

````

Some advice to improve 

* Do read the doc of the tools you use
* Try to see if what you want to do comes from the tools you are using or from Quarto. 

If you want to use other tools, you also have the **gt** package which has great footnote support as you can target specific cells and it support a syntax to place the footnote on a value

* https://gt.rstudio.com/reference/tab_footnote.html
* Search footnote in **gt** intro: https://gt.rstudio.com/articles/intro-creating-gt-tables.html

Hope it helps

**Comment by kmink3225** [comment]
Thank you for your suggestion, @mcanouil . After reading your response, I realized that I had not fully understood the process of rendering and generating output. Your suggestion provided me with some clues on how to address the overlooked aspects.

Thank you, @cderv , for your incredibly helpful response. Understanding the suggestion provided by @mcanouil was challenging, but thanks to your detailed guidance, I now have a better grasp of how quarto, markdown, and the kableExtra package work. This opportunity has allowed me to learn specific methods for understanding and utilizing the given resources (quarto, markdown, package) to increase the flexibility in implementing the desired output.

Thanks to both, I feel that my capabilities in R and quarto have advanced to a higher level. Your support in catching fish and teaching how to fish and excel in fishing is greatly appreciated!

**Reply by cderv** [reply]
Glad it helped you ! 

Though, curious who are  Mr. Gil-dong and Mr. Cheol-su ? 🤔

**Reply by kmink3225** [reply]
Haha, I'm sorry. Never mind. It was my mistake (typo). Before I asked you and replied to your answer to my question, I used a Google translator for proofreading and better communication. I just needed fake names for the substitution for two of you, to use the Google translator :).
